### ISSUE TITLE
Define a technical proposal to implement multi-language communities in URLO

### ISSUE THREAD

**Issue by sebasmagri** [rust-community/team#124]

**Comment by KiChjang** [comment]
cc me, @tennix, @wayslog, @3442853561, @zonyitoo for the Chinese community.

**Comment by sebasmagri** [comment]
Quick update. It seems like the best approach for this, given the current limitations of discourse, would be to have a category per language, and then use tags to "sub-categorize".

**Comment by skade** [comment]
Well, but that would mean that each and every user would have to set up default filters.

**Comment by Manishearth** [comment]
Yeah, this is why I strongly prefer individual communities over catch-alls. Catch-alls just get too noisy. It would be nice if it's easy to set up new communities for languages, but I'd prefer if they were separate forums.

**Comment by sebasmagri** [comment]
It'd be less noisy if we setup the landing to use categories instead of topics. What do you think about this?

**Comment by skade** [comment]
@Manishearth seperate forums have a usability (multiple logins needed) and discovery (how do people find out?) problem until discourse supports them. :(

**Comment by Manishearth** [comment]
I mean that you advertise them in the same semi-official way you do /r/rust on the website.

**Comment by skade** [comment]
But that's semi-official. If we want to support multiple languages at our main venue, it should be first-class as all else.

**Comment by gnunicorn** [comment]
Generally, I am fond of the discourse core developer team approach of only creating a category once enough content has been found that justifies to have its own category. That of course, often means if one doesn't find the proper category - especially around language - people tend to not post the content. So I'd propose to create a 'locals' main-category with the explicit recommendation-pinned-post to post in your local language and create sub-categories for each language (or localisation? like, is this to create geo-local groups or language-local groups?) once enough content justifies that. (maybe ask the authors to tag their content with an i18n lang code when posting?)

Regarding noisy-ness: current discourse allows for general muting based on categories (as well as tags) and even allows to set up a default muting for all new users. _IF_ the content in the locals categories does get a lot, people could either mute specific sub-categories or the discourse admins could even make that a default setting. And that, my friends, would be a great problem to have. Because at that point, there is so much content, that then the question to have a fully localised instance of the forum would actually justify the amount of work it requires to maintain it.

**Comment by Manishearth** [comment]
> If we want to support multiple languages at our main venue, it should be first-class as all else.

Sure, let them become official as they grow.


I strongly prefer the model where we let communities grow organically on their own and then make them official if they want to later. E.g. /r/rust has grown on its own. It doesn't want to be official, but if it did I'm sure we'd be fine with that. This apply doubly to other-language-communities that we can't moderate as well.

**Comment by skade** [comment]
Is it also possible to introduce a category so that it is automatically muted for everyone?

I'd be quite okay with a mixed language forum for me.

**Comment by sebasmagri** [comment]
@skade Yes, it's possible to have a category muted so it does not appear in the main page. I've done some tests and we could also setup subcategories. Also, we can set up tags per category. So, I think we can reach an adequate solution using this tools having hidden categories per language and probably a pinned topic as an index with links to the language specific categories.

If everybody agrees with this approach, I'd like to setup a test with an URLO mirror. Would it be possible?

Best Regards,

**Comment by skade** [comment]
Does anyone know how the mozilla-community discourse works? They have a nice overview page in front of everything.

**Comment by johannhof** [comment]
@skade there are people listed as module peer and owner at https://wiki.mozilla.org/Community_Ops/Discourse

I personally don't know them but maybe someone can catch them on IRC :)

**Comment by sebasmagri** [comment]
It seems like the Mozilla Community Discourse instance uses the previously described approach for multi-language: https://discourse.mozilla-community.org/

**Comment by notriddle** [comment]
You can solve the discoverability problem by having language links on top of the page. The same way https://talk.commonmark.org/ has links to the spec and stuff, https://users.rust-lang.org/ can have a drop-down menu to pick your preferred language.

**Comment by sebasmagri** [comment]
I've contacted a person I know who happens to be in the group of module peers for Mozilla Community's discourse... I'll let you know of any feedback.
